server/services/zwavejs-ui/lib/zwaveJSUI.setValue.js

```diff
@@ -1,35 +1,38 @@
 const get = require('get-value');
 const { BadParameters } = require('../../../utils/coreErrors');
-const { COMMANDS } = require('./constants');
-const { cleanNames } = require('../utils/convertToGladysDevice');
+const { ACTIONS } = require('./constants');
+const { cleanNames, getDeviceFeatureId } = require('../utils/convertToGladysDevice');
 
 /**
- * @description Returns the command wrapper.
+ * @description Returns the action wrapper.
+ * @param {object} zwaveJsNode - The zWaveJsDevice node.
  * @param {object} nodeFeature - The feature.
- * @returns {object} The Command Class command.
+ * @returns {object} The Command Class action.
  * @example
- * getCommand({command_class_name: 'Notification', property: 'Home Security', property_key: 'Cover Status'})
+ * getAction(
+ *  {id: 5, deviceClass: { basic: 4, generic: 17, specific: 6}},
+ *  {command_class_name: 'Notification', property: 'Home Security', property_key: 'Cover Status'}
+ * )
  */
-function getCommand(nodeFeature) {
-  let commandPath = `${cleanNames(nodeFeature.command_class_name)}.${cleanNames(nodeFeature.property_name)}`;
+function getAction(zwaveJsNode, nodeFeature) {
+  let baseCommandPath = cleanNames(nodeFeature.property_name);
   const propertyKeyNameClean = cleanNames(nodeFeature.property_key_name);
   if (propertyKeyNameClean !== '') {
-    commandPath += `.${propertyKeyNameClean}`;
+    baseCommandPath += `.${propertyKeyNameClean}`;
   }
 
-  return get(COMMANDS, commandPath);
-}
+  if (nodeFeature.feature_name) {
+    baseCommandPath += `.${nodeFeature.feature_name}`;
+  }
 
-/**
- * @description Returns a node from its external id.
- * @param {Array} nodes - All nodes available.
- * @param {string} nodeId - The node to find.
- * @returns {object} The node if found.
- * @example
- * getNode([{external_id: 'zwavejs-ui:3'}], 'zwavejs-ui:3')
- */
-function getNode(nodes, nodeId) {
-  return nodes.find((n) => n.external_id === nodeId);
+  return (
+    get(
+      ACTIONS,
+      `${cleanNames(nodeFeature.command_class_name)}.${zwaveJsNode.deviceClass.generic}-${
+        zwaveJsNode.deviceClass.specific
+      }.${baseCommandPath}`,
+    ) || get(ACTIONS, `${cleanNames(nodeFeature.command_class_name)}.${baseCommandPath}`)
```

Comment:
Maybe let's put this in a util function?

Reply:
Moved to a function